packages/next-app/src/app/stableswapsimulator/(components)/StableCurve.tsx

```diff
@@ -1,32 +1,164 @@
+import { useState } from "react";
+import Plot from "react-plotly.js";
+
+import { Spinner } from "#/components/Spinner";
 import { useStableSwap } from "#/contexts/StableSwapContext";
 
 export default function StableCurve() {
   const {
     baselineData,
-    variantData,
     indexAnalysisToken,
     indexCurrentTabToken,
+    preparePoolPairData,
+    numberToOldBigNumber,
   } = useStableSwap();
-  const amountsAnalysisToken = calculateAmounts(
-    baselineData?.tokens?.[indexAnalysisToken]?.balance
+
+  if (
+    !baselineData ||
+    !baselineData.swapFee ||
+    !baselineData.ampFactor ||
+    !baselineData.tokens
+  )
+    return <Spinner />;
+
+  const [formattedPoints, setFormattedPoints] = useState<{
+    x: string;
+    y: string;
+  }>();
+
+  const tokensSymbol = baselineData.tokens.map((token) => token.symbol);
+
+  const initialAmountsAnalysisTokenIn = calculateAmounts({
+    balance: baselineData?.tokens?.[indexAnalysisToken]?.balance,
+  });
+
+  const initialAmountsAnalysisTokenOut = initialAmountsAnalysisTokenIn.map(
+    (amount) => -1 * amount //This is being multiplied by -1 because it is the OUT part of the trade
   );
-  const amountsTabToken = calculateAmounts(
-    variantData?.tokens?.[indexAnalysisToken]?.balance
+
+  const initialPoolPairDataAnalysisIn = preparePoolPairData({
+    indexIn: indexAnalysisToken,
+    indexOut: indexCurrentTabToken,
+    swapFee: baselineData?.swapFee,
+    allBalances: baselineData?.tokens?.map((token) => token.balance),
+    amp: baselineData?.ampFactor,
+  });
+
+  const initialAmountTabTokenOut = initialAmountsAnalysisTokenIn.map(
+    (amount) => {
+      return (
+        StableMath._exactTokenInForTokenOut(
+          numberToOldBigNumber(amount),
+          initialPoolPairDataAnalysisIn
+        ).toNumber() * -1 //This is being multiplied by -1 because it is the OUT part of the trade
+      );
+    }
   );
+
+  const initialPoolPairDataAnalysisOut = preparePoolPairData({
+    indexIn: indexCurrentTabToken,
+    indexOut: indexAnalysisToken,
+    swapFee: baselineData?.swapFee,
+    allBalances: baselineData?.tokens?.map((token) => token.balance),
+    amp: baselineData?.ampFactor,
+  });
+
+  const initialAmountTabTokenIn = initialAmountsAnalysisTokenIn.map(
+    (amount) => {
+      return StableMath._exactTokenInForTokenOut(
+        numberToOldBigNumber(amount),
+        initialPoolPairDataAnalysisOut
+      ).toNumber();
+    }
+  );
+
   return (
-    <>
-      {amountsAnalysisToken} {amountsTabToken} {indexCurrentTabToken}
-    </>
+    <div className="text-white">
+      <Plot
+        data={[
+          {
+            x: initialAmountsAnalysisTokenIn,
+            y: initialAmountTabTokenOut,
+            type: "scatter",
+            mode: "lines",
+            marker: { color: blueDarkA.blueA9 },
+            hovertemplate: `Swap $%{x:.2f} ${tokensSymbol[indexAnalysisToken]} for ${formattedPoints?.y} ${tokensSymbol[indexCurrentTabToken]} <extra></extra>`,
```

Comment:
we should think of a way to round those values (depending on the scale). E.g. 10100.321-> 10.1k, 13.32->13.32.

Reply:
[Issue for this](https://linear.app/bleu-llc/issue/BAL-421/hover-decimals-depending-of-pool-scale)